e2etests/tests/unnumbered.go

```diff
@@ -0,0 +1,286 @@
+		remoteASN := rc.ASNLocal
+		prefixSend := []string{"5.5.5.0/24", "5555::/64"}
+
+		externalP2PContainer, err := frrcontainer.CreateP2PPeerFor(nodeWithP2PConnection.Name, iface, FRRImage)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.By(fmt.Sprintf("create p2p %s:%s -- %s:%s)", nodeWithP2PConnection.Name, iface, iface, externalP2PContainer.Name))
+		ginkgo.DeferCleanup(func() {
+			err := frrcontainer.Delete([]*frrcontainer.FRR{externalP2PContainer})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		ginkgo.By(fmt.Sprintf("load frrconfig to %s", externalP2PContainer.Name))
+		c, err := rc.Config()
+		Expect(err).NotTo(HaveOccurred())
+		err = externalP2PContainer.UpdateBGPConfigFile(c)
+		Expect(err).NotTo(HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("apply frrconfiguration %s", "for-"+externalP2PContainer.Name))
+		cr := frrk8sv1beta1.FRRConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "for-" + externalP2PContainer.Name,
+				Namespace: k8s.FRRK8sNamespace,
+			},
+			Spec: frrk8sv1beta1.FRRConfigurationSpec{
+				BGP: frrk8sv1beta1.BGPConfig{
+					BFDProfiles: []frrk8sv1beta1.BFDProfile{{Name: "simple"}},
+					Routers: []frrk8sv1beta1.Router{
+						{
+							ASN: infra.FRRK8sASN,
+							Neighbors: []frrk8sv1beta1.Neighbor{
+								{
+									ASN:        remoteASN,
+									Interface:  iface,
+									BFDProfile: "simple",
+									ToReceive: frrk8sv1beta1.Receive{
+										Allowed: frrk8sv1beta1.AllowedInPrefixes{
+											Mode: frrk8sv1beta1.AllowAll,
+										},
+									},
+									ToAdvertise: frrk8sv1beta1.Advertise{
+										Allowed: frrk8sv1beta1.AllowedOutPrefixes{
+											Mode: frrk8sv1beta1.AllowAll,
+										},
+									},
+								},
+							},
+							Prefixes: prefixSend,
+						},
+					},
+				},
+				NodeSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"kubernetes.io/hostname": nodeWithP2PConnection.GetLabels()["kubernetes.io/hostname"],
+					},
+				},
+			},
+		}
+		err = updater.Update([]corev1.Secret{}, cr)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := updater.DeleteFRRConfiguration(cr)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		nodeLLA, err := container.GetIPv6Link(nodeWithP2PConnection.Name, iface)
+		Expect(err).NotTo(HaveOccurred())
+		peerLLA, err := container.GetIPv6Link(externalP2PContainer.Name, iface)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.By(fmt.Sprintf("%s/%s/%s --- %s/%s/%s",
+			nodeWithP2PConnection.Name, iface, nodeLLA, iface, peerLLA, externalP2PContainer.Name))
+
+		validateBGPPeering(externalP2PContainer, nodeLLA)
+
+		ginkgo.By(fmt.Sprintf("validate routing table on %s (prefixSend)", externalP2PContainer.Name))
+		validatePrefixViaFor(externalP2PContainer, iface, nodeLLA, prefixSend...)
+		// we cannot use
+		// ValidatePrefixesForNeighbor(*externalP2PContainer, []corev1.Node{nodeWithP2PConnection}, prefixSend...)
+		// because deep down the LLA ip cannot be found as part of the node
```

Comment:
remove from here, put a similar comment on `validatePrefixViaFor` (explaining why we need to use this and not `ValidatePrefixesForNeighbor` as usual)

Reply:
i think done